cli/tests/services.bats

```diff
@@ -339,59 +352,41 @@ EOF
 @test "watchdog: lives as long as the activation" {
   export FLOX_FEATURES_SERVICES=true
   setup_sleeping_services
-  run "$FLOX_BIN" activate -s -- bash <(cat <<'EOF'
+  export -f watchdog_pids_called_with_arg
+  run --separate-stderr "$FLOX_BIN" activate -- bash <(cat <<'EOF'
     source "${TESTS_DIR}/services/register_cleanup.sh"
 
-    log_file="$PWD/.flox/cache/$(ls .flox/cache)"
-
     # Ensure that the watchdog is still running
-    if tail -n 1 "$log_file" | grep "exiting"; then
-      exit 1
-    fi
+    times=0
+    while true; do
+      if [ "$times" -gt 100 ]; then
+        exit 1
+      fi
+      pid="$(watchdog_pids_called_with_arg $_FLOX_SERVICES_SOCKET)"
+      if [ -n "${pid?}" ]; then
+        echo "$pid"
+        break
+      fi
+      times=$((times + 1))
+      sleep 0.01
+    done
 EOF
 )
+  pid="$output"
   assert_success
 
   # Ensure that the watchdog has exited now
-  log_file="$PWD/.flox/cache/$(ls .flox/cache)"
-  if ! tail -n 1 "$log_file" | grep "exiting"; then
-    exit 1
-  fi
-}
-
-@test "watchdog: exits on termination signal (SIGUSR1)" {
```

Comment:
(blocking) Do we still want this test after reverting the `kill -0` approach and keeping the SIGUSR1 behaviour?

Reply:
Yes, this is what I get for reverting commits that weren't perfectly atomic

Reply:
Fixed